### PR TITLE
added team overview functionality

### PR DESCRIPTION
**Pull request by PaulTummonUnosquare**

### CODE REVIEW

File: BackEnd/.classpath [hunk 1]
```diff
@@ -27,5 +27,10 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry kind="src" path="target/generated-sources/annotations">
+		<attributes>
+			<attribute name="optional" value="true"/>
+		</attributes>
+	</classpathentry>
```
Comment (William-Unosquare) [inline BackEnd/.classpath]: Not sure this is supposed to be here, could you remove it from the PR?

File: web-app/src/pages/Admin/index.jsx [hunk 2]
```diff
@@ -34,6 +35,7 @@ export const Admin = props => (
           path="/admin/clients/"
           render={() => <AllClients history={props.history} />}
         />
+        <Route path="/admin/teamOverview" component={TeamOverview} />
```
Comment (William-Unosquare) [inline web-app/src/pages/Admin/index.jsx]: A small thing, but try clean up this URL to be similar to the others. Something like:
`/admin/overview/team`, or even `/admin/overview/`

File: web-app/src/components/TeamOverview/container.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      super(props);
+      this.state = {
+        teams: [],
+        selectedUser: {},
+      };
+    }
+
+    componentDidMount() {
+      
+      var date = new Date().toISOString().slice(0,10);
```
Comment (William-Unosquare) [inline web-app/src/components/TeamOverview/container.js]: Try to use moment dates. I can go over what you should do here instead to follow convention.

File: web-app/src/components/DataTable/Cells/teamsOverview.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  accessor: user => user.team,
+};
+  
+  
+export default {
+  fullName,
+  state,
+  team,
+};
+
```
Comment (William-Unosquare) [inline web-app/src/components/DataTable/Cells/teamsOverview.js:25]: Add a newline at the end of the file. I'd recommend something like Prettier to do this for you when you save.

**Review by William-Unosquare — CHANGES_REQUESTED** [review]
A couple of changes here, but nice work. You put this together well! I'd consider renaming this feature a bit. It's not immediately clear what exactly it is.

File: web-app/src/components/TeamOverview/container.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      
+      getTeamOverview( date )
+        .then( ({data}) => this.setState( { teams: data } ) )
```
Comment (ghost) [inline web-app/src/components/TeamOverview/container.js]: Really small but could you remove the space between .`then( ({data})`

File: web-app/src/components/TeamOverview/container.js (hunk 3 above)
Comment (ghost) [reply web-app/src/components/TeamOverview/container.js]: I know you'll be changing this but try to use const, let etc.

File: web-app/src/components/index.js [hunk 6]
```diff
@@ -22,3 +22,6 @@ export { default as HolidayModal } from './HolidayModal';
 export { default as AllEmployees } from './AllEmployees';
 export { default as DataTable } from './DataTable';
 export { default as FullPageLoader } from './FullPageLoader';
+export { default as TeamOverview } from './TeamOverview';
+
+
```
Comment (ghost) [inline web-app/src/components/index.js:27]: Small comment! But could you remove the extra line here, apologies to be pedantic :)

File: web-app/src/components/TeamOverview/container.js (hunk 3 above)
Comment (PaulTummonUnosquare) [reply web-app/src/components/TeamOverview/container.js]: yeah no problem,....thought I had it as const, sorry about that. : )

Comment (ghost) [reply web-app/src/components/TeamOverview/container.js]: Easy done :)

**Review by ghost — DISMISSED** [review]
Looking good so far, minor tweaks stylistically etc but looking good so far.

**Comment by William-Unosquare** [comment]
There's still a lot of back-end files in this branch.

File: web-app/src/components/TeamOverview/container.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    constructor(props) {
+      super(props);
+      this.state = {
+        teams: [],
+        selectedUser: {},
+      };
+    }
+
+    componentDidMount() {
+
```
Comment (William-Unosquare) [inline web-app/src/components/TeamOverview/container.js:21]: Remove space

File: web-app/src/components/TeamOverview/container.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    componentDidMount() {
+      
+      getTeamOverview( )
+        .then( ({data}) => this.setState( { teams: data } ))
+        .catch(error =>
+          Swal('Error', `There was an error: ${error.message}`, 'error')
+        );
+
```
Comment (William-Unosquare) [inline web-app/src/components/TeamOverview/container.js:27]: Remove space

File: web-app/src/components/TeamOverview/container.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    selectUser = user => this.setState({ selectedUser: user });
+
+    render() {
+      return (
+        <Wrapped
+          history={this.props.history}
+          teams={this.state.teams}
+          selectedUser={this.state.selectedUser}
```
Comment (William-Unosquare) [inline web-app/src/components/TeamOverview/container.js]: De-structure state

File: web-app/src/components/TeamOverview/index.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        columns={['fullName', 'state', 'team']}
+        onRowClick={employee => selectUser(employee)}
+        pageSize={20}
+      />
+    </Fragment>
+  );
+};
+
+TeamOverview.propTypes = {
+  teams: PT.array,
```
Comment (William-Unosquare) [inline web-app/src/components/TeamOverview/index.jsx:28]: When something isn't required try set up a default prop

File: web-app/src/utilities/AxiosInstance.js [hunk 11]
```diff
@@ -55,6 +55,20 @@ instance.interceptors.response.use(function(response) {
     };
   }
 
+  if (response.config.url.includes(`${baseURL}/dashboard/getDashboardSnapshot`)) {
+
```
Comment (William-Unosquare) [inline web-app/src/utilities/AxiosInstance.js]: space

**Review by William-Unosquare — CHANGES_REQUESTED** [review]
More spaces and some other clean up :P Looks great!

**Review by William-Unosquare — APPROVED** [review]

**Comment by ghost** [comment]
Looks good, once those styling things are sorted I'll approve. Nicely done Sir :)

**Pull request merged**